docker/iroh-fedimintd/deploy.sh

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+# domain to use for a host
+domain="$1"
+# ssh host name to use for ssh commands (defaults to the domain above)
+ssh_host="${2:-${domain}}"
+
+# host dir we'll keep the files on
+host_dir=/root/fedimint-docker
+
+##
+## Wipe the previous host
+##
+
+echo >&2 '### Wiping previous setup'
+
+# shellcheck disable=SC2087
+ssh -q "root@$ssh_host" << EOF
+  touch ~/.hushlogin
+
+  systemctl stop fedimint-docker-compose 2>/dev/null || true
+
+  if [ -e $host_dir  ] ; then
+    # Stop the docker-compose stack
+    cd $host_dir && docker-compose down 2>/dev/null || true
+    # We remove only the fedimintd_data named volume
+    docker volume rm fedimint-docker_fedimintd_data 2>/dev/null || true
+  fi
+
+  rm -rf /root/fedimint-docker
+  rm -rf /etc/systemd/system/fedimint-docker-compose.service
+  sudo systemctl daemon-reload
+
+EOF
+
+
+##
+## Setup new machine
+##
+
+echo >&2 '### Setting up the server'
+
+echo >&2 '### Copying files...'
+cat << EOF | ssh "root@$ssh_host" "sudo cat > /etc/systemd/system/fedimint-docker-compose.service"
+[Unit]
+Description=Fedimint Docker Compose Service
+Requires=docker.service
+After=docker.service
+
+[Service]
+WorkingDirectory=${host_dir}
+ExecStart=/usr/bin/docker-compose up
+ExecStop=/usr/bin/docker-compose down
+Restart=always
+
+[Install]
+WantedBy=multi-user.target
+EOF
+
+scp -r .env docker-compose.yaml "root@$ssh_host:/root/fedimint-docker"
+
+echo >&2 '### Setting up...'
+
+# shellcheck disable=SC2087
+ssh -q "root@$ssh_host" << EOF
+  touch ~/.hushlogin
+
+  ufw --force reset
```

Comment:
This is the primary difference from the other `deploy.sh` script, which allows ports 80, 443, and 22. In this script, we reset the `ufw` rules entirely because Iroh I've postponed unifying this with the existing deploy script to avoid disrupting the Hetzner boxes currently used for release testing. This script specifically handles deployments to my self-hosted infrastructure.

Reply:
The benefit of keeping the `deploy.sh` and `setup.sh` scripts is for testing with my own infra. I'll maintain this logic on my end, no need to introduce this complexity to `fedimint/fedimint`.